File: persephone/provisioning-service/src/main/java/com/vmware/blockchain/deployment/services/provision/DeploymentSessionContext.java
/* **************************************************************************
 * Copyright (c) 2019 VMware, Inc. All rights reserved. VMware Confidential
 * **************************************************************************/

package com.vmware.blockchain.deployment.services.provision;

import java.util.Map;

import javax.validation.constraints.NotNull;

import com.vmware.blockchain.deployment.services.orchestration.Orchestrator;
import com.vmware.blockchain.deployment.v1.OrchestrationSiteIdentifier;

/**
 * Data object which stored session.
 */
@Deprecated
public class DeploymentSessionContext {

    @NotNull
    private final Map<OrchestrationSiteIdentifier, Orchestrator> orchestrators;

    @NotNull
    public final Map<OrchestrationSiteIdentifier, Orchestrator> getOrchestrators() {
        return this.orchestrators;
    }

    public DeploymentSessionContext(@NotNull Map<OrchestrationSiteIdentifier, Orchestrator> orchestrators) {
        this.orchestrators = orchestrators;
    }
}

File: persephone/provisioning-service/src/jvm/com/vmware/blockchain/deployment/service/provision/DeploymentSessionContext.kt
/* **************************************************************************
 * Copyright (c) 2019 VMware, Inc.  All rights reserved. VMware Confidential
 * *************************************************************************/
package com.vmware.blockchain.deployment.service.provision

import com.vmware.blockchain.deployment.orchestration.Orchestrator
import com.vmware.blockchain.deployment.v1.OrchestrationSiteIdentifier

/**
 * Context encapsulating the runtime parameters and states of a running deployment session.
 */
data class DeploymentSessionContext(
    val orchestrators: Map<OrchestrationSiteIdentifier, Orchestrator>
)
